frontend/src/metabase/visualizations/lib/apply_axis.js

```diff
@@ -116,9 +116,6 @@ export function applyChartTimeseriesXAxis(
       chart.settings["graph.x_axis.gridLine_enabled"],
     );
 
-    if (dimensionColumn.unit == null) {
-      dimensionColumn = { ...dimensionColumn, unit: dataInterval.interval };
```

Comment:
This approach created a race condition since calls to `isValid` for settings are cached. If tick were formatted before settings/tooltips are shown, it would constantly return `false` because of the invalid `unit` value.